### PR TITLE
Feature/d3 asim 2842

### PR DESCRIPTION
**Pull request by lana-shanghai**

### CODE REVIEW

File: src/d3a/models/area/__init__.py [hunk 1]
```diff
@@ -198,11 +198,18 @@ def _set_grid_fees(self, transfer_fee_const, grid_fee_percentage):
...
+            return grid_fee_constant + self.parent.get_path_to_root_fees()
+        else:
+            return self.grid_fee_constant if self.grid_fee_constant else 0
+
     def get_grid_fee(self):
         grid_fee_type = self.config.grid_fee_type \
             if self.config is not None \
             else ConstSettings.IAASettings.GRID_FEE_TYPE
-        return self.grid_fee_constant if grid_fee_type == 1 else self.grid_fee_percentage
+        return self.get_path_to_root_fees(self) if grid_fee_type == 1 else self.grid_fee_percentage
```
Comment (faizan2590) [inline src/d3a/models/area/__init__.py:212]: Can you please confirm with Sarah if percentage based grid fee also needs to be taken here?

File: integration_tests/fees_along_path.feature [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        Given we have a scenario named grid_fees/test_grid_fees
+        And d3a is installed
+        When we run the simulation with setup file grid_fees.test_grid_fees and parameters [24, 60, 60, 1]
+        Then the load initial buying rate adds grid fees
+
+    Scenario: 
+        Given we have a scenario named grid_fees/test_grid_fees
+        And d3a is installed
+        When we run the simulation with setup file grid_fees.test_grid_fees and parameters [24, 60, 60, 1]
+        Then the PV initial selling rate subtracts grid fees
```
Comment (faizan2590) [inline integration_tests/fees_along_path.feature:13]: I think these two scenarios could be squashed into one simply via adding the step `the PV initial selling rate subtracts grid fees` after L7 at this point of commit

Comment (lana-shanghai) [reply integration_tests/fees_along_path.feature:13]: Would it start with "And" or "Then"?

Comment (faizan2590) [reply integration_tests/fees_along_path.feature:13]: All the following conditions would start with `And`

File: integration_tests/steps/integration_tests.py [hunk 3]
```diff
@@ -1080,6 +1080,50 @@ def identical_profiles(context):
...
 
+@then("the PV initial selling rate subtracts grid fees")
+def pv_selling_rate_minus_fees(context):
+    grid = context.simulation.area
+    hood2 = list(filter(lambda x: x.name == "Neighborhood 2", context.simulation.area.children))[0]
+    house2 = list(filter(lambda x: x.name == "House 2", hood2.children))[0]
+    pv = list(filter(lambda x: x.name == "H2 PV", house2.children))[0]
+
+    market_maker_rate = 30
+    fees_path_to_root = 1 + 2
```
Comment (faizan2590) [inline integration_tests/steps/integration_tests.py:1091]:
```suggestion
    fees_path_to_root = hood2.grid_fee_constant + house2.grid_fee_constant
```

**Review by faizan2590 — DISMISSED** [review]
Left some enhancement comments, once addressed, LGTM.

**Review by faizan2590 — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**